### PR TITLE
Ft integrate requests trips 168781688

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

Allows a user to view requests that they have made on the barefoot nomad platform from over a period of time.

#### Description of Task to be completed?

- Allow a user to view their requests by hitting an endpoint.
- Create relations between the requests table, the locations table, the destinations & the request statuses.

#### How should this be manually tested?

1. Clone this repo
2. Run `npm install `to install all dependencies
3. Run ` sequelize db:migrate ` to migrate all the tables
4. Run ` sequelize db:seed:all ` to seed the default values for locations, requestStatus  & tripTypes tables.  
5. Sign up to the platform / login - take note of the id & token properties in the data object of the response
6. Manually create a record in the requests table. (Default accomodation to the value 1). Refer to the seeded location table for locationId value and the seeded tripTypes table for typeId value.
7. Manually create a record in the destination table. Refer to seeded tables for the corresponding Id values. 
8. With the token in the header [bearer token], use postman to make a ` GET ` request to ` localhost:3000/api/v1/requests ` 
9. Verify that the correct data is returned.

#### Any background context you want to provide?

N / A

#### Screenshots (if appropriate)

Screen Shot 2019-10-16 at 11 04 23
Screen Shot 2019-10-16 at 11 04 30
Screen Shot 2019-10-16 at 11 04 46
Screen Shot 2019-10-16 at 11 05 01

#### Questions:
N / A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]

**Review by alainmateso — CHANGES_REQUESTED** [review]
Great Job Cheza! One concern though, why not seed all this information that is instructured to be created manually? I think they should be seeded as well!

**Comment by Cheza-Dzabala** [comment]
The reason for not seeding is because some seeder may be used in production while others may not. I was thinking to only seed information that is supposed to be available by default and manually enter information that is supposed to be inputted by the user.

**Comment by Cheza-Dzabala** [comment]
> Great Job Cheza! One concern though, why not seed all this information that is instructured to be created manually? I think they should be seeded as well!

**Review by sabin18 — APPROVED** [review]
good job @Cheza-Dzabala .

File: src/controllers/requestController.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import userServices from '../services/userServices';
+import Utilities from '../utils/index';
+
+export default class requestController {
+  static async viewRequests({ user }, res) {
+    const query = Utilities.requestQueries.userRequests(user.payload);
+    const { requests } = await userServices.findOne(query, Utilities.queryScopes.responseScope);
```
Comment (nakiwuge) [inline src/controllers/requestController.js:8]: Add a new line

File: src/routes/api/requests.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express';
+import requestController from '../../controllers/requestController';
+import validateToken from '../../middlewares/auth/validateToken';
+
+const router = new Router();
+
+
```
Comment (nakiwuge) [inline src/routes/api/requests.js]: Remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job on the implementation. Kindly work on returning the id for `origin,location,status and request type`.

I have also left a few comments you should check out

**Review by nakiwuge — APPROVED** [review]

**Review by alainmateso — APPROVED** [review]

**Review by nignanthomas — APPROVED** [review]
@Cheza-Dzabala 
Reviewed and approved 👍

**Pull request merged**